test/e2e/performanceprofile/functests/2_performance_update/updating_profile.go

```diff
@@ -806,6 +819,364 @@ var _ = Describe("[rfe_id:28761][performance] Updating parameters in performance
+				profile.Spec.WorkloadHints = &performancev2.WorkloadHints{
+					PerPodPowerManagement: pointer.BoolPtr(true),
+					HighPowerConsumption:  pointer.BoolPtr(false),
+					RealTime:              pointer.BoolPtr(true),
+				}
+				By("Patching the performance profile with workload hints")
+				workloadHints, err := json.Marshal(profile.Spec.WorkloadHints)
+				Expect(err).ToNot(HaveOccurred())
+
+				Expect(testclient.Client.Patch(context.TODO(), profile,
+					client.RawPatch(
+						types.JSONPatchType,
+						[]byte(fmt.Sprintf(`[{ "op": "replace", "path": "/spec/workloadHints", "value": %s }]`, workloadHints)),
+					),
+				)).ToNot(HaveOccurred())
+				By("Applying changes in performance profile and waiting until mcp will start updating")
+				mcps.WaitForCondition(performanceMCP, machineconfigv1.MachineConfigPoolUpdating, corev1.ConditionTrue)
+
+				By("Waiting when mcp finishes updates")
+				mcps.WaitForCondition(performanceMCP, machineconfigv1.MachineConfigPoolUpdated, corev1.ConditionTrue)
+
+				annotations := map[string]string{
+					"cpu-c-states.crio.io":      "enable",
+					"cpu-freq-governor.crio.io": "schedutil",
+				}
+
+				cpuCount := "2"
+				resCpu := resource.MustParse(cpuCount)
+				resMem := resource.MustParse("100Mi")
+				testpod = pods.GetTestPod()
+				testpod.Namespace = testutils.NamespaceTesting
+				testpod.Spec.Containers[0].Resources = corev1.ResourceRequirements{
+					Limits: corev1.ResourceList{
+						corev1.ResourceCPU:    resCpu,
+						corev1.ResourceMemory: resMem,
+					},
+				}
+				testpod.Spec.NodeSelector = map[string]string{testutils.LabelHostname: workerRTNodes[0].Name}
+				testpod.Annotations = annotations
+				runtimeClass := components.GetComponentName(profile.Name, components.ComponentNamePrefix)
+				testpod.Spec.RuntimeClassName = &runtimeClass
+
+				By("creating test pod")
+				err = testclient.Client.Create(context.TODO(), testpod)
+				Expect(err).ToNot(HaveOccurred())
+				err = pods.WaitForCondition(testpod, corev1.PodReady, corev1.ConditionTrue, 10*time.Minute)
+				Expect(err).ToNot(HaveOccurred())
+				Expect(testpod.Status.QOSClass).To(Equal(corev1.PodQOSGuaranteed), "Test pod does not have QoS class of Guaranteed")
+
+				By("Getting the container cpuset.cpus cgroup")
+				containerID, err := pods.GetContainerIDByName(testpod, "test")
+				Expect(err).ToNot(HaveOccurred())
+
+				containerCgroup := ""
+				Eventually(func() string {
+					cmd := []string{"/bin/bash", "-c", fmt.Sprintf("find /rootfs/sys/fs/cgroup/cpuset/ -name *%s*", containerID)}
+					containerCgroup, err = nodes.ExecCommandOnNode(cmd, &workerRTNodes[0])
+					Expect(err).ToNot(HaveOccurred())
+					return containerCgroup
+				}, (cluster.ComputeTestTimeout(30*time.Second, RunningOnSingleNode)), 5*time.Second).ShouldNot(BeEmpty(),
+					fmt.Sprintf("cannot find cgroup for container %q", containerID))
+
+				By("Checking what CPU the pod is using")
```

Comment:
Maybe it is too much, but I wonder if it would be useful to check that cpus not in use by the pod did not have their pm_qos_resume_latency_us or scaling_governor modified?

Reply:
ack will check that

Reply:
@bartwensley i have addressed this in the latest patch